vm/ubpf_jit_x86_64.h

```diff
@@ -319,11 +319,22 @@ emit_store_imm32(struct jit_state *state, enum operand_size size, int dst, int32
 static inline void
 emit_call(struct jit_state *state, void *target)
 {
+#if defined(_WIN32)
+    /* Windows x64 ABI requires home register space */
+    /* Allocate home register space - 4 registers */
```

Comment:
Why not put the emit_push() here (and the emit_pop at the end)?

Reply:
Good point.

Reply:
I was originally only going to do this if there are 5 parameters. Given that I changed it to always do it (to keep the ubpf_register function simpler), I guess it makes sense to do it here.

Note:
We will need to modify ubpf_register to support x86 JIT (assuming we ever do) as it needs to know the exact size of of parameters to adjust the stack correctly).

Reply:
I see. This is going to be a performance hit for every call, but I don't know if there is an easy way in ubpf to know how many parameters a function has.

Reply:
We could modify:
https://github.com/iovisor/ubpf/blob/843394752c752e38f79d6d49fc704a73ee6490bd/vm/inc/ubpf.h#L67

To accept a parameter count, store that in ubpf_vm and pass parameter count to:
 https://github.com/iovisor/ubpf/blob/843394752c752e38f79d6d49fc704a73ee6490bd/vm/ubpf_jit_x86_64.h#L320

But until I get more data on how bad the perf hit is, I was thinking of keeping the code simpler. 

Note:
This only affects uBPF on Windows. While Linux does spill 7th+ parameter to the stack, BPF only defines 5 parameters for BPF calls :-)